MODIFY: roleSelector

triblox opened the pull request:
The selection of gaming roles is now handled in private message

triblox committed:
MODIFY: roleSelector

The selection of gaming roles is now handled in private

JulesdeCube commented on botygeek/commands/roles.py:
```diff
@@ -22,17 +22,27 @@ async def on_member_join(self, member : discord.Member):
-      if str(addrole) == rolename:
-        await ctx.author.add_roles(addrole)
+    if isinstance(ctx.channel, discord.DMChannel):
+      await ctx.send("test")
```
no

JulesdeCube commented on botygeek/commands/roles.py:
```diff
@@ -22,17 +22,27 @@ async def on_member_join(self, member : discord.Member):
+    if isinstance(ctx.channel, discord.DMChannel):
+      await ctx.send("test")
+      for id in self.config.roleSelector:
+        theguild = self.bot.get_guild(self.config.guildID)
```
create a propoerty and not try to load it every times

JulesdeCube commented on botygeek/commands/roles.py:
```diff
@@ -22,17 +22,27 @@ async def on_member_join(self, member : discord.Member):
+        addrole = theguild.get_role(id)
+        if str(addrole) == rolename:
+          theauthor = ctx.message.author
+          authoringuild = theguild.get_member_named(str(theauthor))
```
not use pseudo but id and there are probably a methode to have it

JulesdeCube commented on botygeek/commands/roles.py:
```diff
@@ -22,17 +22,27 @@ async def on_member_join(self, member : discord.Member):
+        if str(addrole) == rolename:
+          theauthor = ctx.message.author
+          authoringuild = theguild.get_member_named(str(theauthor))
+          await authoringuild.add_roles(addrole)
```
check if in the server

JulesdeCube commented on botygeek/commands/roles.py:
```diff
@@ -22,17 +22,27 @@ async def on_member_join(self, member : discord.Member):
-      removerole = ctx.guild.get_role(id)
-      if str(removerole) == rolename:
-        await ctx.author.remove_roles(removerole)
+    if isinstance(ctx.channel, discord.DMChannel):
+      await ctx.send("test")
+      for id in self.config.roleSelector:
+        theguild = self.bot.get_guild(self.config.guildID)
+        addrole = theguild.get_role(id)
+        if str(addrole) == rolename:
+          theauthor = ctx.message.author
+          authoringuild = theguild.get_member_named(str(theauthor))
+          await authoringuild.remove_roles(addrole)
```
same as before (create a function to get the user in the server we can need it after

triblox committed:
MODIFY: roleSelector

add some embed and modify request pull change

triblox committed:
MODIFY: enrror embed message

JulesdeCube commented on botygeek/commands/roles.py:
```diff
@@ -67,7 +77,10 @@ async def removerole_error(self, ctx : commands.Context, error : commands.Comman
       return
     self.log(LogType.LogType.ERROR, "REMOVEROLE/ERROR", error)
-
+  
+  @staticmethod
+  def get_player_by_ID(self, author_id : int):
+    return self.theguild.get_member(author_id)
```
why not use direcly the user and not the id

JulesdeCube replied:
why not use the user direcly as parameter ?

JulesdeCube commented on botygeek/logs/ClientLogs.py:
```diff
@@ -79,6 +79,15 @@ async def helpMessage(ctx : discord.abc.Messageable, name : str, description : s
+  async def actionDone(ctx : discord.abc.Messageable, name : str, description : str, arguments = None) -> discord.Message:
+    embed : discord.Embed = discord.Embed(color=LogColor[LogType.VALIDATE])
+    embed.set_author(icon_url=LogIcon[LogType.VALIDATE], name=str.capitalize(name))
+    embed.description = description
     if arguments:
       embed.add_field(name="Arguments", value=str.join("\n - ", arguments), inline=False)
     return await ctx.send(embed=embed)
```
why a Argument field ?

JulesdeCube committed:
MODIFY : change getrole to addrole ADD: role error

ADD:
- `Roles` : add `roleSelector` and `roleOnJoin`proerty to not load it
  every time, add error message when user is not in the server and the
  role didn't exist
- `Config.py` :add type of `roleOnJoin` and `roleSelector`

MODIFY :
- global : remove space at end of lines
- `README.MD` : change `getrole` to  `addrole` and guildID to
  `serverID`
- `Config.py` : rename `guildID` to `serverID`
- `ClientLogs.py` : add posibility to set the title in `send` methode

JulesdeCube committed:
FIX : removerole add role and didn't remove it
